File: C/beginner/1061.c
// https://judge.beecrowd.com/pt/problems/view/1061
// Tempo de um evento

// Libraries
#include <stdio.h>

// Function to convert days, hours and minutes to seconds
int convert(int days, int hours, int minutes, int seconds) {
    return (days * 24 * 60 * 60) + (hours * 60 * 60) + (minutes * 60) + seconds;
}

// Main Function
int main(){
    // Variables
    int s_day, s_hour, s_minute, s_second;
    int e_day, e_hour, e_minute, e_second;

    // Scanning values
    scanf("Dia %d\n", &s_day);
    scanf("%d : %d : %d\n", &s_hour, &s_minute, &s_second);
    scanf("Dia %d\n", &e_day);
    scanf("%d : %d : %d\n", &e_hour, &e_minute, &e_second);

    // Doing the calcs to find time
    int starting = convert(s_day, s_hour, s_minute, s_second);
    int ending = convert(e_day, e_hour, e_minute, e_second);
    int duration = ending - starting;

    // Showing results
    printf("%d dia(s)\n", duration / (24 * 60 * 60));
    duration %= 24 * 60 * 60;
    printf("%d hora(s)\n", duration / (60 * 60));
    duration %= 60 * 60;
    printf("%d minuto(s)\n", duration / 60);
    duration %= 60;
    printf("%d segundo(s)\n", duration);

    // Ending Code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra
/* Special thanks to xTecna on https://github.com/xTecna who made the original code an gave me ideas
to made this code. */

File: Haskell/beginner/1061.hs
-- https://judge.beecrowd.com.br/judge/pt/problems/view/1061
-- Event Duration

import Text.Printf (printf)

-- Function to convert days, hours, minutes, and seconds to total seconds
convert :: Int -> Int -> Int -> Int -> Int
convert days hours minutes seconds = 
    (days * 24 * 60 * 60) + (hours * 60 * 60) + (minutes * 60) + seconds

-- Main function
main :: IO ()
main = do
    -- Reading the start day
    s_input <- getLine  -- Example: "Dia 5"
    let s_day = read (last (words s_input)) :: Int  -- Extract the number after "Dia"

    -- Reading the start time (hours, minutes, seconds)
    timeStr1 <- getLine
    let [s_hour, s_minute, s_second] = map read (words (map (\c -> if c == ':' then ' ' else c) timeStr1)) :: [Int]

    -- Reading the end day
    e_input <- getLine  -- Example: "Dia 9"
    let e_day = read (last (words e_input)) :: Int  -- Extract the number after "Dia"

    -- Reading the end time (hours, minutes, seconds)
    timeStr2 <- getLine
    let [e_hour, e_minute, e_second] = map read (words (map (\c -> if c == ':' then ' ' else c) timeStr2)) :: [Int]

    -- Calculating the total time in seconds
    let starting = convert s_day s_hour s_minute s_second
    let ending = convert e_day e_hour e_minute e_second
    let duration = ending - starting

    -- Displaying the result
    printf "%d Dia(s)\n" (duration `div` (24 * 60 * 60))
    let duration' = duration `mod` (24 * 60 * 60)
    printf "%d Hora(s)\n" (duration' `div` (60 * 60))
    let duration'' = duration' `mod` (60 * 60)
    printf "%d Minuto(s)\n" (duration'' `div` 60)
    printf "%d Segundo(s)\n" (duration'' `mod` 60)


-- By Fernando Serra
-- https://github.com/fernandosserra
-- Most of the code was built with the help of Chat GPT and the Codeium extension